components/script/dom/url.rs

```diff
@@ -77,38 +77,34 @@ impl URL {
         url: USVString,
         base: Option<USVString>,
     ) -> Fallible<DomRoot<URL>> {
-        let parsed_base = match base {
-            None => {
-                // Step 1.
-                None
-            },
-            Some(base) =>
-            // Step 2.1.
-            {
-                match ServoUrl::parse(&base.0) {
-                    Ok(base) => Some(base),
-                    Err(error) => {
-                        // Step 2.2.
-                        return Err(Error::Type(format!("could not parse base: {}", error)));
-                    },
-                }
-            },
-        };
-        // Step 3.
-        let parsed_url = match ServoUrl::parse_with_base(parsed_base.as_ref(), &url.0) {
-            Ok(url) => url,
-            Err(error) => {
-                // Step 4.
-                return Err(Error::Type(format!("could not parse URL: {}", error)));
-            },
+        // Step 1
+        let parsed_url = {
+            let parsed_base = match base {
+                Some(base) => {
+                    match ServoUrl::parse(&base.0) {
+                        Ok(base) => Some(base),
+                        Err(error) => {
+                            // Step 2
+                            return Err(Error::Type(format!("could not parse base: {}", error)));
+                        },
+                    }
+                },
+                None => None,
+            };
+            match ServoUrl::parse_with_base(parsed_base.as_ref(), &url.0) {
+                Ok(url) => url,
+                Err(error) => {
+                    // Step 2
+                    return Err(Error::Type(format!("could not parse URL: {}", error)));
+                },
+            }
         };
-        // Step 5: Skip (see step 8 below).
-        // Steps 6-7.
-        let result = URL::new(global, parsed_url);
-        // Step 8: Instead of construcing a new `URLSearchParams` object here, construct it
-        //         on-demand inside `URL::SearchParams`.
```

Comment:
I don't know exactly why this happened, but this PR doesn't change the behavior around `URLSearchParams`, continuing to ignore the related steps.

Reply:
Is this change just updating the step numbers?

Reply:
Yes. The corresponding spec change related to this PR can be found here: https://github.com/whatwg/url/commit/ae3c28b8

(I probably should have mentioned this somewhere, such as in commit message.)

Reply:
I'm not sure the code changes are an improvement here, because they increase the nesting level of the method. Would it be possible to avoid changing the code if there is no behavior changes and also to paste a small description of each step in quotes. With a small amount of text, at least it will still be clear what each step does.

Reply:
Okay, that makes sense. I'll make the change.